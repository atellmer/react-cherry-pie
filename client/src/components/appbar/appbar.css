@import '../../variables.css';


:local(.root) {
	position: relative;
	display: flex;
	flex-flow: row wrap;
	align-items: center;
	justify-content: space-between;
	padding: 0 16px;
	min-height: 42px;
	width: 100%;
	height: auto;
	background-color: var(--primaryColorFirst);
	box-shadow: var(--shadowElevationZ1);
	transition: height .2s ease-in-out;

	@media (min-width: 600px) {
		min-height: 48px;
	}
}

:local(.logoLayout) {
	@media (min-width: 600px) {
		display: none;
	}
}

:local(.avatarLayout) {
	margin: 2px;
}

:local(.iconButtonLayout) {
	margin: 0 -16px 0 0;

	button {
		@media (max-width: 600px) {
			padding: 9px!important;
			width: 42px!important;
			height: 42px!important;
		}
	}
}
