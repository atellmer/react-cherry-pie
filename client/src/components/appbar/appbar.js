/** @flow */
import React, { Component } from 'react';
import { Flex, Box } from 'reflexbox';
import Avatar from 'material-ui/Avatar/index';
import IconButton from 'material-ui/IconButton/index';
import MoreVertIcon from 'material-ui/svg-icons/navigation/more-vert';

import { FakeUserService } from '../../utils/fakeUsers';
import css from './appbar.css';
import TmLogo from '../logo';

type Props = {};
type State = {
	userAvatar: string
}

class TmAppbar extends Component {
	props: Props;
	state: State;
	fakeUserService: FakeUserService;
	userAvatar: string;

	constructor(props: Props) {
		super(props);
		this.state = {
			userAvatar: ''
		};
		this.fakeUserService = new FakeUserService();
	}

	componentWillMount() {
		this.fakeUserService
			.getFakeUser()
			.subscribe(res => {
				this.setState({
					userAvatar: res.results[0].picture.thumbnail
				});
			});
	}

	render() {
		return (
			<div className={css.root}>
				<div>
					<div className={css.logoLayout}>
						<TmLogo/>
					</div>
				</div>
				<Flex align='center'>
					<Box className={css.avatarLayout}>
						<Avatar src={this.state.userAvatar} size={34} backgroundColor={'transparent'}/>
					</Box>
					<Box className={css.iconButtonLayout}>
						<IconButton>
							<MoreVertIcon color={'#fff'}/>
						</IconButton>
					</Box>
				</Flex>
			</div>
		);
	}
}

export default TmAppbar;
