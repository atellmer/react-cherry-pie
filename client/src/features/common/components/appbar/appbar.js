/** @flow */
import React from 'react';
import { Link } from 'react-router-dom';
import { pure } from 'recompose';
import Avatar from 'material-ui/Avatar/index';
import IconButton from 'material-ui/IconButton/index';
import ArrowBackIcon from 'material-ui/svg-icons/navigation/arrow-back';

import AppbarMenu from '../appbarMenu';
import {
  Logo
} from '@/features/ui';
import {
  PROFILE_ROUTE,
  MESSENGER_ROUTE
} from '@/vars';
import type {
  UserType
} from '@/features/common';
import {
  Root,
  ArrowBackLayout,
  LogoLayout,
  ControlsLayout,
  AvatarLayout,
  IconButtonLayout
} from './styled';


type Props = {
  me: UserType,
  user: any,
  signout: Function
};

const Appbar = (props: Props) => {
  const renderBackBtn = () => {
    const {
      user
    } = props;

    if (user) {
      return (
        <ArrowBackLayout>
          <Link
            to={`/${MESSENGER_ROUTE}`}>
            <IconButton>
              <ArrowBackIcon
                color={'#fff'} />
            </IconButton>
          </Link>
        </ArrowBackLayout>
      );
    }

    return null;
  };

  const renderControls = () => {
    const {
      me: { avatar: { thumbnail: avatar } },
      user,
      signout
    } = props;

    if (user) {
      return (
        <ControlsLayout>
          <AvatarLayout>
            <Link
              to={`/${PROFILE_ROUTE}`}>
              <Avatar
                src={avatar}
                size={34}
                backgroundColor={'transparent'} />
            </Link>
          </AvatarLayout>
          <IconButtonLayout>
            <AppbarMenu
              signout={signout} />
          </IconButtonLayout>
        </ControlsLayout>
      );
    }

    return null;
  };


  return (
    <Root>
      {renderBackBtn()}
      <LogoLayout>
        <Logo />
      </LogoLayout>
      {renderControls()}
    </Root>
  );
};

export { Appbar };
export default pure(Appbar);
