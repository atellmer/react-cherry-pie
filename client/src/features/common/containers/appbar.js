/** @flow */
import React, { Component } from 'react';
import { connect } from 'react-redux';
import { pure, compose } from 'recompose';

import type { UserType } from '@/features/common';
import { Appbar } from '@/features/common';


type Props = {
  me: UserType,
};

class AppbarContainer extends Component<void, Props, *> {

  render() {
    const { me } = this.props;

    return <Appbar me={me}/>;
  }
}

function mapStateToProps({ common }) {
  const { user: { me } } = common;

  return {
    me
  };
}

export default compose(
  pure,
  connect(mapStateToProps),
)(AppbarContainer);
