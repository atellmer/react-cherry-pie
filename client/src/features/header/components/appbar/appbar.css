@import 'variables.css';

:local(.root) {
  position: relative;
  display: flex;
  flex-flow: row wrap;
  align-items: center;
  justify-content: space-between;
  padding: 0 16px;
  min-height: 42px;
  width: 100%;
  height: auto;
  background-color: var(--primaryColorFirst);
  box-shadow: var(--shadowElevationZ1);
  transition: height .2s ease-in-out;

  @media (min-width: 600px) {
    min-height: 48px;
  }
}

:local(.avatarLayout) {
  margin: 2px;
}

:local(.controlsLayout) {
  display: flex;
  flex-flow: row wrap;
  justify-content: center;
  align-items: center;
}

:local(.iconButtonLayout) {
  margin: 0 -16px 0 0;

  button {
    @media (max-width: 600px) {
      padding: 9px !important;
      width: 42px !important;
      height: 42px !important;
    }
  }
}

:local(.arrowBackLayout) {
  margin: 0 0 0 -16px;

  button {
    @media (max-width: 600px) {
      padding: 9px !important;
      width: 42px !important;
      height: 42px !important;
    }
  }
}
