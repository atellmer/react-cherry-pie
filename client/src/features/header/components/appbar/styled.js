/** @flow */
import styled from 'styled-components';

import * as vars from '@/vars';


const Root = styled.div`
  position: relative;
  display: flex;
  flex-flow: row wrap;
  align-items: center;
  justify-content: space-between;
  padding: 0 16px;
  min-height: 42px;
  width: 100%;
  height: auto;
  background-color: ${vars.primaryColorFirst};
  box-shadow: ${vars.shadowElevationZ1};
  transition: height .2s ease-in-out;

  @media (min-width: ${vars.phone}px) {
    min-height: 48px;
  }
`;

const ArrowBackLayout = styled.div`
  margin: 0 0 0 -16px;

  button {
    @media (max-width: ${vars.phone}px) {
      padding: 9px !important;
      width: 42px !important;
      height: 42px !important;
    }
  }
`;

const ControlsLayout = styled.div`
  display: flex;
  flex-flow: row wrap;
  justify-content: center;
  align-items: center;
`;

const AvatarLayout = styled.div`
  margin: 2px;
`;

const IconButtonLayout = styled.div`
  margin: 0 -16px 0 0;

  button {
    @media (max-width: ${vars.phone}px) {
      padding: 9px !important;
      width: 42px !important;
      height: 42px !important;
    }
  }
`;

export {
  Root,
  ArrowBackLayout,
  ControlsLayout,
  AvatarLayout,
  IconButtonLayout
};
