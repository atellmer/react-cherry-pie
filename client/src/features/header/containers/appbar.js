/** @flow */
import React, { Component } from 'react';
import { connect } from 'react-redux';
import { pure, compose } from 'recompose';

import type { IUser } from '@/shared/models/user';
import Appbar from '../components/appbar';


type Props = {
  me: IUser,
};

class AppbarContainer extends Component {
  props: Props;

  render() {
    const { me } = this.props;

    return <Appbar me={me}/>;
  }
}

function mapStateToProps({ user }) {
  const { me } = user;

  return {
    me
  };
}

export default compose(
  pure,
  connect(mapStateToProps),
)(AppbarContainer);
