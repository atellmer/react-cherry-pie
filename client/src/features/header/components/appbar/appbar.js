/** @flow */
import React, { Component } from 'react';
import { pure } from 'recompose';
import Avatar from 'material-ui/Avatar/index';
import IconButton from 'material-ui/IconButton/index';
import MoreVertIcon from 'material-ui/svg-icons/navigation/more-vert';

import type { IUser } from '@/shared/models/user';
import TmLogo from '@/features/ui/logo';
import * as css from './appbar.css';


type Props = {
  me: IUser
};

class TmAppbar extends Component {
  props: Props;

  constructor(props: Props) {
    super(props);
  }

  render() {
    const avatar = this.props.me.avatar.thumbnail;

    return (
      <div className={css.root}>
        <div>
          <div className={css.logoLayout}>
            <TmLogo/>
          </div>
        </div>
        <div className={css.controlsLayout}>
          <div className={css.avatarLayout}>
            <Avatar
              src={avatar}
              size={34}
              backgroundColor={'transparent'}/>
          </div>
          <div className={css.iconButtonLayout}>
            <IconButton>
              <MoreVertIcon color={'#fff'}/>
            </IconButton>
          </div>
        </div>
      </div>
    );
  }
}

export default pure(TmAppbar);
