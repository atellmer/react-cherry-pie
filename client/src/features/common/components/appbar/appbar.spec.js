import React from 'react';
import { shallow } from 'enzyme';
import Avatar from 'material-ui/Avatar/index';

import { Appbar } from './appbar';
import AppbarMenu from '../appbarMenu';


describe('Appbar Component', () => {
  let props;

  beforeEach(() => {
    props = {
      me: {
        avatar: {
          thumbnail: 'path'
        }
      },
      user: {},
      signout: jest.fn()
    };
  });

  test('should render component', () => {
    const component = shallow(<Appbar {...props}/>);

    expect(component).toMatchSnapshot();
  });

  test('should render component without controls', () => {
    const component =  shallow(
      <Appbar
        me={props.me}
        signout={props.signout}/>
    );

    expect(component).toMatchSnapshot();
  });

  test('should render avatar', () => {
    const component = shallow(<Appbar {...props}/>);
    const avatar = component.find(Avatar);

    expect(component.instance().props.me.avatar.thumbnail).toEqual('path');
    expect(avatar.props().src).toEqual('path');
  });

  test('pass callback to appbarMenu', () => {
    const component = shallow(<Appbar {...props}/>);
    const appbarMenu = component.find(AppbarMenu);

    expect(appbarMenu.props().signout).toEqual(props.signout);
  });
});
