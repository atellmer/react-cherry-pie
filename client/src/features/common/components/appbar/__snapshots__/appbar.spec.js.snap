// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Appbar Component should render component 1`] = `
<styled__Root>
  <styled__ArrowBackLayout>
    <Link
      replace={false}
      to="/im"
    >
      <IconButton
        disableTouchRipple={false}
        disabled={false}
        iconStyle={Object {}}
        tooltipPosition="bottom-center"
        touch={false}
      >
        <NavigationArrowBack
          color="#fff"
        />
      </IconButton>
    </Link>
  </styled__ArrowBackLayout>
  <styled__LogoLayout>
    <Logo />
  </styled__LogoLayout>
  <styled__ControlsLayout>
    <styled__AvatarLayout>
      <Link
        replace={false}
        to="/profile"
      >
        <Avatar
          backgroundColor="transparent"
          size={34}
          src="path"
        />
      </Link>
    </styled__AvatarLayout>
    <styled__IconButtonLayout>
      <pure(AppbarMenu)
        signout={[Function]}
      />
    </styled__IconButtonLayout>
  </styled__ControlsLayout>
</styled__Root>
`;

exports[`Appbar Component should render component without controls 1`] = `
<styled__Root>
  <styled__LogoLayout>
    <Logo />
  </styled__LogoLayout>
</styled__Root>
`;
